parser/parser.go

```diff
@@ -85,8 +84,127 @@ func GetParser(fileType string) (Parser, error) {
 	case "xml":
 		return &xml.Parser{}, nil
 	default:
-		return nil, fmt.Errorf("unknown filetype given: %v", fileType)
+		return nil, fmt.Errorf("unknown file extension given: %v", fileExtension)
+	}
+}
+
+// GetParserFromPath returns a file parser based on the file type
+// that exists at the given path.
+func GetParserFromPath(path string) (Parser, error) {
+	fileType := getFileType(path)
+
+	return GetParser(fileType)
+}
+
+// ParseConfigurations parses and returns the configurations from the given
+// list of files.
+func ParseConfigurations(files []string) (map[string]interface{}, error) {
+	configurations, err := parseConfigurations(files, "")
+	if err != nil {
+		return nil, fmt.Errorf("get configurations: %w", err)
+	}
+
+	return configurations, nil
+}
+
+// ParseConfigurationsAs parses the files as the given file type and returns the
+// configurations given in the file list.
+func ParseConfigurationsAs(files []string, fileExtension string) (map[string]interface{}, error) {
+	configurations, err := parseConfigurations(files, fileExtension)
+	if err != nil {
+		return nil, fmt.Errorf("parse configurations: %w", err)
+	}
+
+	return configurations, nil
+}
+
+// CombineConfigurations takes the given configurations and combines them into a single
+// configuration.
+func CombineConfigurations(configs map[string]interface{}) map[string]interface{} {
+	type configuration struct {
+		Path     string      `json:"path"`
+		Contents interface{} `json:"contents"`
+	}
+
+	var allConfigurations []configuration
+	for path, config := range configs {
+		if subconfigs, exist := config.([]interface{}); exist {
+			for _, subconfig := range subconfigs {
+				configuration := configuration{
+					Path:     path,
```

Comment:
I really like this solution! It makes it a lot more obvious how to write policies in the combined case!